src/main/webapp/app/overview/submission-result-status.component.ts

```diff
@@ -46,9 +47,20 @@ export class SubmissionResultStatusComponent implements OnChanges {
             const quizExercise = this.exercise as QuizExercise;
             this.uninitialized = ArtemisQuizService.isUninitialized(quizExercise);
             this.quizNotStarted = ArtemisQuizService.notStarted(quizExercise);
+            this.shouldShowResult = !!this.studentParticipation?.results?.length;
         } else {
             this.uninitialized = !afterDueDate && !this.studentParticipation;
-            this.notSubmitted = !!this.studentParticipation && !this.studentParticipation.submissions?.length;
+            this.notSubmitted = afterDueDate && !!this.studentParticipation && !this.studentParticipation.submissions?.length;
+
+            if (this.exercise.type === ExerciseType.PROGRAMMING) {
+                const initializationStatesForResult = [InitializationState.INITIALIZED, InitializationState.INACTIVE, InitializationState.FINISHED];
+                this.shouldShowResult =
+                    (!!this.studentParticipation?.results?.length || !afterDueDate) &&
+                    !!this.studentParticipation?.initializationState &&
+                    initializationStatesForResult.includes(this.studentParticipation.initializationState);
```

Comment:
```suggestion            initializationStatesForResult.includes(this.studentParticipation?.initializationState);
```
Is the non-`null`/`undefined` assertion needed here? I’d expect `….includes(undefined)` to be `false` anyway.

Reply:
No, in the previous line, we check that the state is truthful. if it weren't, the line before would  be false and hence the last line would not even get evaluated.

Reply:
Typescript complains about undefined if I leave this out.
`Argument of type 'InitializationState | undefined' is not assignable to parameter of type 'InitializationState'.   Type 'undefined' is not assignable to type 'InitializationState'.`